qa/tasks/vstart_runner.py

```diff
@@ -203,8 +203,14 @@ def finished(self):
 
         if self.subproc.poll() is not None:
             out, err = self.subproc.communicate()
-            self.stdout.write(out)
-            self.stderr.write(err)
+            if isinstance(self.stdout, StringIO):
+                self.stdout.write(out.decode(errors='ignore'))
+            else:
+                self.stdout.write(out)
+            if isinstance(self.stderr, StringIO):
+                self.stderr.write(err.decode(errors='ignore'))
+            else:
+                self.stderr.write(err)
```

Comment:
This should be using 'six.ensure_str' instead of inventing a wheel.

Reply:
probably we can use `six` only for py2 compatible?

Reply:
well... it is for good of backporting, by the way I have no idea why this change in this PR if it is not a try to be py2 compatible? 
Btw, in py3
```
     |      communicate() returns a tuple (stdout, stderr).  These will be
     |      bytes or, if self.universal_newlines was True, a string.
```
Which makes it know for sure what will be returned and there is no need to check output types.

Reply:
We are dropping the py2 and only py3 will be support.

Reply:
> We are dropping the py2 and only py3 will be support.

if so, there is no need to use StringIO check stdout, see the comment above.

Reply:
Okay, will check and try this.